test/http-api/inject/test-pubsub.js

```diff
@@ -0,0 +1,139 @@
+/* eslint max-nested-callbacks: ["error", 8] */
+/* eslint-env mocha */
+'use strict'
+
+const expect = require('chai').expect
+const createTempNode = require('./../../utils/temp-node')
+
+module.exports = (http) => {
+  describe('/pubsub', () => {
+    let api
+    let tmpNode
+
+    const buf = new Buffer('some message')
+    const topic = 'nonScents'
+    const topicNotSubscribed = 'somethingRandom'
+
+    before((done) => {
+      api = http.api.server.select('API')
+
+      createTempNode(47, (err, _ipfs) => {
+        expect(err).to.not.exist
+        tmpNode = _ipfs
+        tmpNode.goOnline((err) => {
+          expect(err).to.not.exist
+          done()
+        })
+      })
+    })
+
+    after((done) => {
+      setTimeout(() => {
+        tmpNode.goOffline(done)
+      }, 1000)
+    })
+
+    describe('/sub', () => {
+      it('returns 500 if no topic is provided', (done) => {
+        api.inject({
+          method: 'GET',
+          url: `/api/v0/pubsub/sub`
+        }, (res) => {
+          expect(res.statusCode).to.equal(500)
+          done()
+        })
+      })
+
+      it('returns 200 with topic', (done) => {
+        // TODO: Agree on a better way to test this (currently this hangs)
+        // Regarding: https://github.com/ipfs/js-ipfs/pull/644#issuecomment-267687194
+        // Current Patch: Subscribe to a topic so the other tests run as expected
```

Comment:
The issue is that the the http-api is designed as a "hanging" request if you subscribe to a topic, which is a scenario not really testable with the hapijs inject approach.